crates/runtime/tests/models/openai.rs

```diff
@@ -195,6 +200,88 @@ async fn openai_search_test() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[tokio::test]
+async fn openai_embeddings_test() -> Result<(), anyhow::Error> {
+    let _tracing = init_tracing(None);
+
+    verify_env_secret_exists("SPICE_OPENAI_API_KEY")
+        .await
+        .map_err(anyhow::Error::msg)?;
+
+    let app = AppBuilder::new("search_app")
+        .with_embedding(get_openai_embeddings(
+            Some("text-embedding-3-small"),
+            "openai_embeddings",
+        ))
+        .build();
+
+    let http_port = rand::thread_rng().gen_range(50000..60000);
+
+    tracing::debug!("Running Spice runtime with http port: {http_port}");
+
+    let api_config = Config::new().with_http_bind_address(SocketAddr::new(LOCALHOST, http_port));
```

Comment:
You'll need to specify all of the ports - otherwise the ones you don't specify get their default values, which will conflict in other tests.

Reply:
Updated, thank you